src/Compilers/CSharp/Test/Semantic/Semantics/NameOfTests.cs

```diff
@@ -2373,5 +2366,497 @@ class Attr : System.Attribute { public Attr(string s) {} }";
+        public void OpenTypeInNameof_GenericMethod1()
+        {
+            CompileAndVerify("""
+                using System;
+                    
+                var v = nameof(IGoo.M);
+                Console.WriteLine(v);
+
+                interface IGoo
+                {
+                    void M<T>();
+                }
+                """, expectedOutput: "M").VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void OpenTypeInNameof_GenericMethod2()
+        {
+            CreateCompilation("""
+                using System;
+                    
+                var v = nameof(IGoo.M<>);
+                Console.WriteLine(v);
+
+                interface IGoo
+                {
+                    void M<T>();
+                }
+                """).VerifyDiagnostics(
+                    // (3,16): error CS0305: Using the generic method group 'M' requires 1 type arguments
+                    // var v = nameof(IGoo.M<>);
+                    Diagnostic(ErrorCode.ERR_BadArity, "IGoo.M<>").WithArguments("M", "method group", "1").WithLocation(3, 16));
+        }
+
+        [Fact]
+        public void OpenTypeInNameof_GenericMethod3()
+        {
+            CreateCompilation("""
+                using System;
+                    
+                var v = nameof(IGoo.M<int>);
+                Console.WriteLine(v);
+
+                interface IGoo
+                {
+                    void M<T>();
+                }
+                """).VerifyDiagnostics(
+                // (3,16): error CS8084: Type parameters are not allowed on a method group as an argument to 'nameof'.
+                // var v = nameof(IGoo.M<int>);
+                Diagnostic(ErrorCode.ERR_NameofMethodGroupWithTypeParameters, "IGoo.M<int>").WithLocation(3, 16));
+        }
+
+        [Theory]
+        [InlineData("IGoo<>")]
+        [InlineData("IGoo<>.Count")]
+        public void OpenTypeInNameof_SemanticModelTest1(string nameofType)
+        {
+            var compilation = CreateCompilation($$"""
+                using System;
+                    
+                var v1 = nameof({{nameofType}});
+                var v2 = typeof(IGoo<>);
+                Console.WriteLine(v1 + v2);
+
+                interface IGoo<T> { public T Count { get; } }
+                """).VerifyDiagnostics();
+            var tree = compilation.SyntaxTrees.Single();
+            var semanticModel = compilation.GetSemanticModel(tree);
+
+            var root = tree.GetRoot();
+
+            var firstGeneric = root.DescendantNodes().OfType<GenericNameSyntax>().First();
+            var lastGeneric = root.DescendantNodes().OfType<GenericNameSyntax>().Last();
+
+            Assert.NotSame(firstGeneric, lastGeneric);
+
+            // Ensure the type inside the nameof is the same as the type inside the typeof.
+            var type1 = semanticModel.GetTypeInfo(firstGeneric).Type;
+            var type2 = semanticModel.GetTypeInfo(firstGeneric).Type;
```

Comment:
>firstGeneric [](http://example.com/codeflow?start=50&length=12)

Crucial typo ;-)
Once fixed, this will show that one will be a definition and one will be substituted...

Reply:
good catch.